api/src/test/java/io/opentelemetry/common/AttributesTest.java

```diff
@@ -253,26 +253,4 @@ void toBuilder() {
     // Original not mutated.
     assertThat(partial).isEqualTo(Attributes.newBuilder().setAttribute("cat", "meow").build());
   }
-
-  @Test
-  void deleteByNull() {
-    Attributes.Builder attributes = Attributes.newBuilder();
-    attributes.setAttribute(stringKey("attrValue"), "attrValue");
-    attributes.setAttribute("string", "string");
-    attributes.setAttribute("long", 10);
-    attributes.setAttribute("double", 1.0);
-    attributes.setAttribute("bool", true);
-    attributes.setAttribute("arrayString", new String[] {"string"});
-    attributes.setAttribute("arrayLong", new Long[] {10L});
-    attributes.setAttribute("arrayDouble", new Double[] {1.0});
-    attributes.setAttribute("arrayBool", new Boolean[] {true});
-    assertThat(attributes.build().size()).isEqualTo(9);
-    attributes.setAttribute(stringKey("attrValue"), null);
-    attributes.setAttribute("string", (String) null);
-    attributes.setAttribute("arrayString", (String[]) null);
-    attributes.setAttribute("arrayLong", (Long[]) null);
-    attributes.setAttribute("arrayDouble", (Double[]) null);
-    attributes.setAttribute("arrayBool", (Boolean[]) null);
-    assertThat(attributes.build().size()).isEqualTo(3);
```

Comment:
should we check that we implement a no-op behavior instead of leaving it undefined?

Reply:
I went back and forth on this. I'm not really sure which we should do, since we're explicitly saying that behavior is undefined. :)

Reply:
If undefined, don't test for it. Some may consider tests as a kind of documentation.

Reply:
that was my thinking as well.

Reply:
I would argue that the specification says "undefined behavior", i.e. each implementation of the spec can do whatever they prefer, but the SDK has a well-defined behavior. IMHO we should verify that we implement correctly the no-op behavior in every point that touches `null` attributes.

Reply:
I've added a test that validates the current behavior, with a clear message that it's not required behavior and would be fine to be changed.

Reply:
As a C++ dev, I'd say that would be "implementation-defined" behavior, not undefined behavior 😀